src/kv/tx_view.h

```diff
@@ -229,29 +229,41 @@ namespace kv
      * whether the iteration should continue (true) or stop (false)
      */
     template <class F>
-    bool foreach(F&& f)
+    void foreach(F&& f)
     {
       // Record a global read dependency.
       tx_changes.read_version = tx_changes.start_version;
       auto& w = tx_changes.writes;
+      bool should_continue = true;
 
-      tx_changes.state.foreach([&w, &f](const K& k, const VersionV<V>& v) {
-        auto write = w.find(k);
+      tx_changes.state.foreach(
+        [&w, &f, &should_continue](const K& k, const VersionV<V>& v) {
+          auto write = w.find(k);
 
-        if ((write == w.end()) && !is_deleted(v.version))
-          return f(k, v.value);
-        return true;
-      });
+          if ((write == w.end()) && !is_deleted(v.version))
+          {
+            should_continue = f(k, v.value);
+          }
+
+          return should_continue;
+        });
 
-      for (auto write = tx_changes.writes.begin();
-           write != tx_changes.writes.end();
-           ++write)
+      if (should_continue)
       {
-        if (!is_deleted(write->second.version))
-          if (!f(write->first, write->second.value))
-            return false;
+        for (auto write = tx_changes.writes.begin();
+             write != tx_changes.writes.end();
+             ++write)
+        {
+          if (!is_deleted(write->second.version))
+            if (!f(write->first, write->second.value))
```

Comment:
Minor: extra `{}` here maybe?

Reply:
Yes: This is the awkward merge with #1221, just reverted to the original for now but I'll merge that to master first and then safely wrap this when integrating this PR.

Reply:
Lol no this was a broken revert completely, hence the test being broken. Oops.